c7n/resources/ec2.py

```diff
@@ -302,6 +302,54 @@ def __call__(self, i):
         return self.operator(map(self.match, volumes))
 
 
+@filters.register('termination-protected')
+class DisableApiTermination(Filter):
+    """EC2 instances with ``disableApiTermination`` attribute set
+
+    Filters EC2 instances with ``disableApiTermination`` attribute set to true.
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: termination-protection-enabled
+            resource: ec2
+            filters:
+              - type: termination-protected
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: termination-protection-NOT-enabled
+            resource: ec2
+            filters:
+              - not:
+                - type: termination-protected
+    """
+
+    schema = type_schema('termination-protected')
+
+    def get_permissions(self):
+        return self.manager.get_resource_manager('ec2').get_permissions()
```

Comment:
given above comment... re perms on this filter. then here. you can do perms = [self.permissions] and then perms.extend(self.manager.get_resource_manager('ec2').get